src/main/java/io/confluent/connect/elasticsearch/ElasticsearchClient.java

```diff
@@ -15,102 +15,568 @@
+          url ->
+              credentialsProvider.setCredentials(
+                  new AuthScope(new HttpHost(url)),
+                  new UsernamePasswordCredentials(config.username(), config.password().value())
+              )
+      );
+      builder.setDefaultCredentialsProvider(credentialsProvider);
+    }
+
+    if (config.isBasicProxyConfigured()) {
+      HttpHost proxy = new HttpHost(config.proxyHost(), config.proxyPort());
+      builder.setProxy(proxy);
+
+      if (config.isProxyWithAuthenticationConfigured()) {
+        credentialsProvider.setCredentials(
+            new AuthScope(proxy),
+            new UsernamePasswordCredentials(config.proxyUsername(), config.proxyPassword().value())
+        );
+      }
+
+      builder.setDefaultCredentialsProvider(credentialsProvider);
+    }
+  }
 
   /**
-   * Gets the JSON mapping for the given index and type. Returns {@code null} if it does not exist.
+   * Configures the client to use SSL if configured.
    *
-   * @param index the index
-   * @param type the type
-   * @throws IOException if the client cannot execute the request
+   * @param builder the HttpAsyncClientBuilder
    */
-  JsonObject getMapping(String index, String type) throws IOException;
+  private void configureSslContext(HttpAsyncClientBuilder builder) {
+    SslFactory kafkaSslFactory = new SslFactory(Mode.CLIENT, null, false);
+    kafkaSslFactory.configure(config.sslConfigs());
+
+    SSLContext sslContext;
+    try {
+      // try AK <= 2.2 first
+      sslContext =
+          (SSLContext) SslFactory.class.getDeclaredMethod("sslContext").invoke(kafkaSslFactory);
+      log.debug("Using AK 2.2 SslFactory methods.");
+    } catch (Exception e) {
+      // must be running AK 2.3+
+      log.debug("Could not find AK 2.2 SslFactory methods. Trying AK 2.3+ methods for SslFactory.");
+
+      Object sslEngine;
+      try {
+        // try AK <= 2.6 second
+        sslEngine = SslFactory.class.getDeclaredMethod("sslEngineBuilder").invoke(kafkaSslFactory);
+        log.debug("Using AK 2.2-2.5 SslFactory methods.");
+
+      } catch (Exception ex) {
+        // must be running AK 2.6+
+        log.debug(
+            "Could not find Ak 2.3-2.5 methods for SslFactory."
+                + " Trying AK 2.6+ methods for SslFactory."
+        );
+        try {
+          sslEngine =
+              SslFactory.class.getDeclaredMethod("sslEngineFactory").invoke(kafkaSslFactory);
+          log.debug("Using AK 2.6+ SslFactory methods.");
+        } catch (Exception exc) {
+          throw new ConnectException("Failed to find methods for SslFactory.", exc);
+        }
+      }
+
+      try {
+        sslContext =
+            (SSLContext) sslEngine.getClass().getDeclaredMethod("sslContext").invoke(sslEngine);
+      } catch (Exception ex) {
+        throw new ConnectException("Could not create SSLContext.", ex);
+      }
+    }
+
+    HostnameVerifier hostnameVerifier = config.shouldDisableHostnameVerification()
```

Comment:
Is this needed for self-signed certificates? Maybe add a comment why are we doing this

Reply:
this section is ported from `JestClient` 

here is the PR that added this functionality https://github.com/confluentinc/kafka-connect-elasticsearch/pull/410

doesn't say the reason we need this, not sure myself